packages/flame/lib/src/components/component.dart

```diff
@@ -243,6 +244,28 @@ class Component {
     }
   }
 
+  Map<Type, InheritedComponent>? _inheritedComponents;
+
+  T? dependOnInheritedComponentOfExactType<T extends InheritedComponent>() {
```

Comment:
Can you please add a description of what this method is supposed to do? The name says "depend on ...", but it doesn't make anything depend on anything, so I was wondering if perhaps it is NYI or just needs a better name.

Reply:
I'm mimicking Flutter's approach to InheritedWidget (https://api.flutter.dev/flutter/widgets/BuildContext/dependOnInheritedWidgetOfExactType.html). Perhaps we can drop the `dependOn` since no method marks it as to update its dependant descendants.

What do you think about, `retreiveInheritedComponentOfExactType`? Is not my favourite name, I'm very open to naming suggestions.

Reply:
I'm happy with `read`, I'll resolve. Feel free to unresolve if the name can be improved.